lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -227,6 +227,8 @@ class DocumentImpl extends NodeImpl {
     }
 
     this._contentType = privateData.options.contentType;
+    this._encoding = privateData.options.encoding;
```

Comment:
I would like to normalize things before we reach the Document level. Per https://dom.spec.whatwg.org/#concept-document-encoding each document has an encoding, and not a characterSet. Can we make that true, and get rid of documentImpl._characterSet?